src/ingest/generation.rs

```diff
@@ -0,0 +1,47 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, Mutex};
+
+use chrono::{Datelike, FixedOffset, Utc};
+
+pub struct SequenceGenerator {
+    counter: AtomicUsize,
+    last_reset_date: Mutex<u32>,
+}
+
+impl SequenceGenerator {
+    pub(crate) fn new() -> Self {
+        Self {
+            counter: AtomicUsize::new(1),
+            last_reset_date: Mutex::new(SequenceGenerator::get_current_date_time()),
+        }
+    }
+
+    pub(crate) fn generate_sequence_number(&self) -> usize {
+        let current_date_time = SequenceGenerator::get_current_date_time();
+        {
+            let mut last_reset_day = self
+                .last_reset_date
+                .lock()
+                .expect("last_reset_date should be exist.");
```

Comment:
I have two questions regarding this line:
- Could you run grammar check for the expect message please?
- Could you help me understand the choice of `expect` here please ? To me it seems like we need to use something like `panic!` according to our guideline.

Reply:
I will change the source code from `Mutex` to `RwLock` as below.
Therefore, there seems to be no need to discuss `expect` and `panic!`.

```
    pub struct SequenceGenerator {
        counter: AtomicUsize,
        last_reset_date: RwLock<u32>,
    }

    pub(crate) async fn generate_sequence_number(&self) -> usize {
        let current_date_time = SequenceGenerator::get_current_date_time();
        {
            let last_reset_day = self.last_reset_date.read().await;
            if *last_reset_day == current_date_time {
                return self.counter.fetch_add(1, Ordering::Relaxed);
            }
        }
        {
            let mut last_reset_day = self.last_reset_date.write().await;
            self.counter.store(1, Ordering::Release);
            *last_reset_day = current_date_time;
        }
        self.counter.fetch_add(1, Ordering::Relaxed)
    }
```